addons/account/models/account_move.py

```diff
@@ -1,349 +1,1869 @@
+        related='company_id.partner_id')
+    company_currency_id = fields.Many2one(string='Company Currency', readonly=True,
+        related='journal_id.company_id.currency_id')
+    currency_id = fields.Many2one('res.currency', store=True, readonly=True, tracking=True, required=True,
+        states={'draft': [('readonly', False)]},
+        string='Currency',
+        default=_get_default_currency)
+    foreign_currency_id = fields.Many2one('res.currency', string='Foreign Currency', # do it in _default_currency_id of account.move.line
+        compute='_compute_foreign_currency_id',
+        help="Technical field used to set the default currency on journal items as their must have a currency_id set only when dealing with foreign currencies.")
+    line_ids = fields.One2many('account.move.line', 'move_id', string='Journal Items', copy=True, readonly=True,
+        states={'draft': [('readonly', False)]})
+    partner_id = fields.Many2one('res.partner', readonly=True, tracking=True,
+        states={'draft': [('readonly', False)]},
+        string='Customer/Vendor')
+    commercial_partner_id = fields.Many2one('res.partner', string='Commercial Entity', store=True, readonly=True,
+        compute='_compute_commercial_partner_id')
+    fiscal_position_id = fields.Many2one('account.fiscal.position', string='Fiscal Position', readonly=True,
+        states={'draft': [('readonly', False)]},
+        help = "Fiscal positions are used to adapt taxes and accounts for particular customers or sales orders/invoices. "
+               "The default value comes from the customer.")
+    amount_untaxed = fields.Monetary(string='Untaxed Amount', store=True, readonly=True, tracking=True,
+        compute='_compute_amount')
+    amount_tax = fields.Monetary(string='Tax', store=True, readonly=True,
+        compute='_compute_amount')
+    amount_total = fields.Monetary(string='Total', store=True, readonly=True,
+        compute='_compute_amount',
+        inverse='_inverse_amount_total')
+    amount_residual = fields.Monetary(string='Amount Due', store=True,
+        compute='_compute_amount')
+    amount_untaxed_signed = fields.Monetary(string='Untaxed Amount Signed', store=True, readonly=True, tracking=True,
+        compute='_compute_amount')
+    amount_tax_signed = fields.Monetary(string='Tax Signed', store=True, readonly=True,
+        compute='_compute_amount')
+    amount_total_signed = fields.Monetary(string='Total Signed', store=True, readonly=True,
+        compute='_compute_amount')
+    amount_residual_signed = fields.Monetary(string='Amount Due Signed', store=True,
+        compute='_compute_amount')
+    to_check = fields.Boolean(string='To Check', default=False,
+        help='If this checkbox is ticked, it means that the user was not sure of all the related informations at the time of the creation of the move and that the move needs to be checked again.')
+    amount_by_group = fields.Binary(string="Tax amount by group",
```

Comment:
what? if it's only used in the invoice report (as the function tends to say), then call the function directly in that report instead of defining a field for that.

Reply:
This field is used since version 12.0 by `web_studio` for the `Totals and Subtotals` block to be d&d to make custom invoice / sale reports. Removing this field is a good idea but it needs some changes JS-side.